crates/turbopack-dev-server/src/http.rs

```diff
@@ -117,13 +121,21 @@ pub async fn process_request_with_content_source(
                 }
 
                 let content = file.content();
-                header_map.insert(
-                    "Content-Length",
-                    hyper::header::HeaderValue::try_from(content.len().to_string())?,
+                // Grab ropereader stream, coerce anyhow::Error to std::io::Error
+                let stream_ext = content
+                    .read()
+                    .into_stream()
+                    .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err));
+
+                let gzipped_stream = tokio_util::io::ReaderStream::new(
```

Comment:
Not all content should be compressed, mainly images except SVG. Compressing binary formats will only increase the file size. Can we do an inspection for the content-type to ensure it's a textual format?

Reply:
Probably yes, could have some basic mime check for the content types.

Reply:
Express's [compression](https://www.npmjs.com/package/compression) middleware uses [compressible](https://www.npmjs.com/package/compressible)

Reply:
yes, that's I figures. But we should have rust corresponding one instead, isn't it?

Reply:
The rust crate seems out of date? We could get most of the way by just using the fallback algorithm from `compressible`:

- text/*
- */*+json
- */*+text
- */*+xml

and a few extras for `application/json` and `application/javascript`

Reply:
You mean only compress for those, correct?

Reply:
Yes